python/ray/data/_internal/shuffle_and_partition.py

```diff
@@ -56,7 +56,7 @@ def map(
         slice_sz = max(1, math.ceil(block.num_rows() / output_num_blocks))
         slices = []
         for i in range(output_num_blocks):
-            slices.append(block.slice(i * slice_sz, (i + 1) * slice_sz, copy=True))
```

Comment:
sorry if I miss anything, why it's safe to remove `copy=True` here?

Reply:
This `copy=True` here (and elsewhere in the PR) was to avoid a smaller-scope version of this Arrow serialization bug in `pyarrow <= 6.0.1` that we observed a while back, where the zero-copy slice views produced by `copy=False` would result in the entire `block` buffer to be sent for every slice. To circumvent this, we eagerly copied the block slices.

With this Arrow serialization hook, this copy (i.e. "buffer truncation", where the zero-copy slice view is turned into a truncated copy of the corresponding slice of the buffer) will happen automatically at serialization time, so we don't need to explicitly copy the slices here.